tests/unit/test-firefox/test.firefox.js

```diff
@@ -977,73 +974,28 @@ describe('firefox', () => {
 
   describe('defaultRemotePortFinder', () => {
 
-    function findRemotePort({...args}: RemotePortFinderParams = {}) {
-      return firefox.defaultRemotePortFinder({...args});
+    function findRemotePort() {
+      return firefox.defaultRemotePortFinder();
     }
 
     it('resolves to an open port', () => {
-      const connectToFirefox = sinon.spy(
+      sinon.spy(
         () => Promise.reject(new TCPConnectError()));
-      return findRemotePort({connectToFirefox})
+      return findRemotePort()
         .then((port) => {
           assert.isNumber(port);
         });
     });
 
-    it('returns a port on first try', () => {
-      const connectToFirefox = sinon.spy(() => new Promise(
-        (resolve, reject) => {
-          reject(
-            new TCPConnectError('first call connection fails - port is free')
-          );
-        }));
-      return findRemotePort({connectToFirefox, retriesLeft: 2})
-        .then((port) => {
-          sinon.assert.calledOnce(connectToFirefox);
-          assert.isNumber(port);
-        });
-    });
-
-    it('cancels search after too many fails', () => {
-      const client = fake(RemoteFirefox.prototype);
-      const connectToFirefox = sinon.spy(() => new Promise(
-        (resolve) => resolve(client)));
-      return findRemotePort({connectToFirefox, retriesLeft: 2})
-        .catch((err) => {
-          assert.equal(err, 'WebExtError: Too many retries on port search');
-          sinon.assert.calledThrice(connectToFirefox);
-        });
-    });
-
-    it('retries port discovery after first failure', () => {
-      const client = fake(RemoteFirefox.prototype);
-      let callCount = 0;
-      const connectToFirefox = sinon.spy(() => {
-        callCount++;
-        return new Promise((resolve, reject) => {
-          if (callCount === 2) {
-            reject(new TCPConnectError('port is free'));
-          } else {
-            resolve(client);
-          }
-        });
-      });
-      return findRemotePort({connectToFirefox, retriesLeft: 2})
-        .then((port) => {
-          assert.isNumber(port);
-          sinon.assert.calledTwice(connectToFirefox);
-        });
-    });
-
     it('throws on unexpected errors', async () => {
-      const connectToFirefox = sinon.spy(async () => {
+      sinon.spy(async () => {
```

Comment:
uhm... this test doesn't seem to be testing anything useful.

To learn more about what is the role of mocha, sinon and chai, you may want to take a look at they docs:

- sinon: https://sinonjs.org/#get-started / https://sinonjs.org/how-to/ https://sinonjs.org/releases/v7.4.1/
- chai: https://www.chaijs.com/api/
- mocha: https://mochajs.org/#getting-started

While learning new concepts and tools is helpful (and often needed) to try them in an isolated test project (believe me on this, I also do it very often, it helps to focus on the new things one at the time and be able to build a mental model of how they work and when/why you will want to use them or not use them ;-))

At a first glance one of the articles linked in the sinon howtos page seems helpful to get started with mocha+sinon+chai (and get an initial picture of how they works):
  
https://scotch.io/tutorials/how-to-test-nodejs-apps-using-mocha-chai-and-sinonjs

Another important thing while writing tests, at least from my point of view, is "keeping an eye on the reasons why you are working on that test" (every test should have a clear role), e.g. asking yourself questions like:

- what is exactly the scenario that we are going to write a test for?
(build a picture / a mental model of it)

- why we have to test that scenario?
(e.g. it makes us sure that a certain behavior doesn't regress, or it may cover some part of the code that is not executed by any of the other tests... etc.)

- is the test going to fail if the code is broken? 
(in my opinion this is one of the most important ones, the tests are not meant to be "green" all the time, they are meant to be "red" when something is broken)

- is the test going to fail intermittently without any code being changed or broken? 
(this is something that becomes more and more important over the time, if you can't trust that a test only fails when something is actually broken, it is not going to be useful enough because we will still need to investigate it manually every time it fails to know if something is actually broken)

Reply:
I read through the docs and they were really helpful! Thank you! :D